src/components/popup/popup.tsx

```diff
@@ -38,6 +45,17 @@ export class Popup extends React.Component<PopupProps, {}> {
         return null;
     }
 
+    public componentDidUpdate() {
+        if (this.isOpenStateChanged) {
```

Comment:
This is problematic on several levels:
1. if I remove isOpenStateChanged from the code no tests are failing
2. open/onOpenStateChange should conform the controlled component pattern: onOpenStateChange should send the value which should be passed by the parent to the prop "open". Then you won't need to check for "isOpenStateChanged"